src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -688,12 +688,11 @@ private static DetailAST getFirstClassNameNode(DetailAST ast) {
     private static boolean isInIgnoreBlock(DetailAST methodBodyAst, DetailAST throwAst) {
         DetailAST ancestor = throwAst;
         while (ancestor != methodBodyAst) {
-            if (ancestor.getType() == TokenTypes.LITERAL_TRY
-                    && ancestor.findFirstToken(TokenTypes.LITERAL_CATCH) != null
```

Comment:
Why was the catch moved to the bottom of the if?

> except it is now the last operand of the or-statement to help mitigate the performance impact by a bit.

What performance impact are you referring to?

Reply:
> What performance impact are you referring to?

Referring to the performance impact that is mentioned in the PR description.
<br>

> Why was the catch moved to the bottom of the if?

Let's say if the type of `ancestor` is either `LAMBDA` or `OBJBLOCK`. 
If the operands were in original ordering, `findFirstToken()` would be the first to run in every iteration regardless the type, checking all the children of the ancestor.
This will all be in vain because it wouldn't find `LITERAL_CATCH` because it can only exist as the child for `ancestor` if the type is `LITERAL_TRY`.
Shifting that to the last, helps us avoid this because it will exit immediately if the type matches either `LAMBDA` or `OBJBLOCK`.

This wouldn't be helpful for cases where the type doesn't matches either `LAMBDA` or `OBJBLOCK` or if the `ancestor` is in some intermediate part of such type. Still some optimization is better than none.

Reply:
I believe you are wrongly assuming one token is more likely to appear than another. If "ancestor is either LAMBDA or OBJBLOCK", but what if ancestor is CATCH then it negates your argument as putting it first would trigger it the fastest. OBJBLOCK is more likely to be seen than any other because more structures use it, but the rarity of LAMBDA versus CATCH is highly dependent on each Java source. One could have a ton of Lambdas or a ton of catches for whichever reason. For example, my projects don't use lambdas but does have some catches.

We can leave it like this for now.

Reply:
True, occurrence of such tokens is code dependent, but my main reason to change the ordering was that the `getType()` is comparatively much faster than `findFirstToken()`.

Reply:
I glossed over `findFirstToken`. I am resolving this now.